### PR TITLE
Fix OOM error in tests when using public Github Runners.

### PR DESCRIPTION
**Pull request by potiuk**
There was a side effect caused by th TestStandardRunner 
test that caused broken logging configuration,
which in turn created OutOfMemory condition for our Public
GitHubRunners.

The problem was that the test overrode the configuration of
logging with some simple test configuration, but never restored
the default configuration, which resulted in `airflow.processor`
logger that was created before contain empty handlers. Since
the `airflow.processor` logger has "propagate" set to False,
empty handlers normally cause a lastResort handler call, which
by default redirects everything to Stderr and this is what
happened in DagFile Processor tests. However, DagFileProcessor
uses `stderr_redirect` which replaces `sys.stderr` with provided
stream. In this case however the stream set (StreamLogWriter)
redirected the output to "airflow.processor" logger - which in
turn (as last resort) redirected everything to sys.stderr which
in turn redirected everything to "airflow.processor" logger etc.

This resulted in:

* OOM condition in Public GitHub Runners
* DagFileProcessor failing with exceeded recursion depth when
  there was enough memory to get there.

The condition was triggered by two preceding tests:

1) First test_plugins_manger.py initialized logging for
   `airflow.processor` and stored it in logging manager
2) The TestStandardTaskRunner test applied simpler configuration
   but the way configure() works - it did not remove the
   "airflow.processor" logger, but it REMOVED all handlers
   registered for it - and never restored the default configuration
3) The DagFileProcessor logs caused infinite recursion

The fix is two-fold:

* the TestStandardTaskRunner restores default config after test
* the DagFileProcessor sets default config before starting

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: tests/dag_processing/test_manager.py [hunk 1]
```diff
@@ -111,6 +113,7 @@ def waitable_handle(self):
 
 class TestDagFileProcessorManager:
     def setup_method(self):
+        dictConfig(DEFAULT_LOGGING_CONFIG)
```
Comment (jedcunningham) [inline tests/dag_processing/test_manager.py:116]: Should we call `logging_config.configure_logging()` instead?

Comment (potiuk) [reply tests/dag_processing/test_manager.py:116]: I copied it from TestStandardTaskRunner - there it is much more obvious why I used it this way. I have no super strong opinion, though I'd say in tests stating expllicitly that "this test requires default config" is a bit more resilient - for example if another test generates local_config and does not remove it - using DEFULT_* will still work.

Comment (jedcunningham) [reply tests/dag_processing/test_manager.py:116]: That's fair. At the end of the day, like normal, the only thing that is consistent is that things are inconsistent 😀.

**Review by jedcunningham — APPROVED** [review]

File: tests/dag_processing/test_manager.py (hunk 1 above)
Comment (potiuk) [reply tests/dag_processing/test_manager.py:116]: Indeed. This PR is a clear manifestation of it :)

**Comment by potiuk** [comment]
Some failiures but the main problem is fixed - we can deal with those separately.

**Comment by kaxil** [comment]
Well done on the PR description too 👏

**Comment by potiuk** [comment]
Thanks @kaxil . The sheer amount of time I spent on it, means that I had to be creative to describe the 4 lines (actually 2 really needed) of code it produced.

**Pull request merged**